common/src/main/java/com/microsoft/identity/common/internal/fido/LegacyFidoActivityResultContract.java

```diff
@@ -0,0 +1,151 @@
+import android.util.Base64;
+
+import androidx.activity.result.IntentSenderRequest;
+import androidx.activity.result.contract.ActivityResultContract;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+import com.google.android.gms.fido.Fido;
+import com.google.android.gms.fido.fido2.api.common.AuthenticatorAssertionResponse;
+import com.google.android.gms.fido.fido2.api.common.AuthenticatorErrorResponse;
+import com.google.android.gms.fido.fido2.api.common.AuthenticatorResponse;
+import com.google.android.gms.fido.fido2.api.common.PublicKeyCredential;
+
+import kotlin.Unit;
+import kotlin.jvm.functions.Function1;
+
+/**
+ * An implementation of an ActivityResultContract which helps run the PendingIntent from the legacy FIDO2 API and connect its results to the FIDO challenge handlers and managers.
+ */
+public class LegacyFidoActivityResultContract extends ActivityResultContract<LegacyFido2ApiObject, Void> {
+    Function1<String, Unit> assertionCallback;
+    Function1<LegacyFido2ApiException, Unit> errorCallback;
+
+    @NonNull
+    @Override
+    public Intent createIntent(@NonNull final Context context, @NonNull final LegacyFido2ApiObject input) {
+        assertionCallback = input.getAssertionCallback();
+        errorCallback = input.getErrorCallback();
+        return new Intent(ACTION_INTENT_SENDER_REQUEST)
+                .putExtra(
+                        EXTRA_INTENT_SENDER_REQUEST,
+                        new IntentSenderRequest.Builder(
+                                input.getPendingIntent().getIntentSender()
+                        ).build());
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
+    @Override
+    public Void parseResult(int resultCode, @Nullable Intent intent) {
+        if (intent == null) {
+            errorCallback.invoke(
+                    new LegacyFido2ApiException(
+                            LegacyFido2ApiException.NULL_OBJECT,
+                            "Result intent from legacy FIDO2 API was null."
+                    ));
+            return null;
+        }
+        if (resultCode != Activity.RESULT_OK) {
+            errorCallback.invoke(
+                    new LegacyFido2ApiException(
+                            LegacyFido2ApiException.BAD_ACTIVITY_RESULT_CODE,
+                            "Activity closed with result code: " + resultCode
+                    ));
+            return null;
+        }
+
+        final byte[] bytes = intent.getByteArrayExtra(Fido.FIDO2_KEY_CREDENTIAL_EXTRA);
+        if (bytes == null) {
+            errorCallback.invoke(
+                    new LegacyFido2ApiException(
+                            LegacyFido2ApiException.NULL_OBJECT,
+                            "Credential result from Intent is null."
+                    ));
+            return null;
+        }
+        final PublicKeyCredential credential = PublicKeyCredential.deserializeFromBytes(bytes);
```

Comment:
nit: extra line after each "return null" block for readability.

Reply:
cef6e97bf8ae3da4badd6b59bcbfed4e63358aba